src/DurableTask/Pageable.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.DurableTask;
+
+/// <summary>
+/// Pageable helpers.
+/// </summary>
+static class Pageable
+{
+    // This code was adapted from Azure SDK AsyncPageable.
+    // TODO: Add Pageable<T> (non-async) when/if it becomes relevant.
+
+    /// <summary>
+    /// Creates an async pageable from a callback function <paramref name="pageFunc" />.
+    /// </summary>
+    /// <typeparam name="T">The type of the value.</typeparam>
+    /// <param name="pageFunc">The callback to fetch additional pages.</param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentNullException"></exception>
+    public static AsyncPageable<T> Create<T>(Func<string?, CancellationToken, Task<Page<T>>> pageFunc)
+        where T : notnull
+    {
+        if (pageFunc is null)
+        {
+            throw new ArgumentNullException(nameof(pageFunc));
+        }
+
+        return Create((continuation, size, cancellation) => pageFunc(continuation, cancellation));
+    }
+
+    /// <summary>
+    /// Creates an async pageable from a callback function <paramref name="pageFunc" />.
+    /// </summary>
+    /// <typeparam name="T">The type of the value.</typeparam>
+    /// <param name="pageFunc">The callback to fetch additional pages.</param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentNullException"></exception>
+    public static AsyncPageable<T> Create<T>(Func<string?, int?, CancellationToken, Task<Page<T>>> pageFunc)
+        where T : notnull
+    {
+        if (pageFunc is null)
+        {
+            throw new ArgumentNullException(nameof(pageFunc));
+        }
+
+        return new FuncAsyncPageable<T>(pageFunc);
+    }
+
+    class FuncAsyncPageable<T> : AsyncPageable<T>
+        where T : notnull
+    {
+        readonly Func<string?, int?, CancellationToken, Task<Page<T>>> pageFunc;
+
+        public FuncAsyncPageable(Func<string?, int?, CancellationToken, Task<Page<T>>> pageFunc)
+        {
+            this.pageFunc = pageFunc;
+        }
+
+        public override IAsyncEnumerable<Page<T>> AsPages(
+            string? continuationToken = default, int? pageSizeHint = default)
+            => this.AsPagesCore(continuationToken, pageSizeHint);
+
+        async IAsyncEnumerable<Page<T>> AsPagesCore(
+            string? continuationToken = default,
+            int? pageSizeHint = default,
+            [EnumeratorCancellation] CancellationToken cancellation = default)
+        {
+            do
+            {
+                Page<T> page = await this.pageFunc(continuationToken, pageSizeHint, cancellation)
+                    .ConfigureAwait(false);
```

Comment:
Same question about the necessity of `.ConfigureAwait(false)`.

Reply:
Same as above - this code has no reliance on any specific context, so `ConfigureAwait(false)` is safe to use. Calling code can still control their own awaiter behavior, allowing them to still be posted back to their own context.